### ISSUE TITLE
Fail to build on riscv32

### ISSUE THREAD

**Issue by Mingli-Yu** [strace/strace#181]
| configure: error: architecture riscv32 is not supported by strace

**Comment by rxrbln** [comment]
we maintain a patch for that in #t2sde, I probably should fill a PR: https://t2sde.org/packages/strace

**Comment by esyr** [comment]
On Thu, Jan 13, 2022 at 8:20 PM René Rebe ***@***.***> wrote:
>
> we maintain a patch for that in #t2sde, I probably should fill a PR: https://t2sde.org/packages/strace

The patch hints that riscv64 does support riscv32 compat, after all.
Is that true?




--
Eugene Syromyatnikov
***@***.***
***@***.***{ru|org}

**Comment by fanghuaqi** [comment]
> we maintain a patch for that in #t2sde, I probably should fill a PR: https://t2sde.org/packages/strace

Hi @rxrbln , will this patch be contributed to this repo, so future buildroot can build it in future release.

**Comment by rxrbln** [comment]
> On Thu, Jan 13, 2022 at 8:20 PM René Rebe ***@***.***> wrote: we maintain a patch for that in #t2sde, I probably should fill a PR: https://t2sde.org/packages/strace
> The patch hints that riscv64 does support riscv32 compat, after all. Is that true?
> […](#)
> -- Eugene Syromyatnikov ***@***.*** ***@***.***{ru|org}

in theory RISCV64 CPUs can support 32-bit user-code. In practice I have not seed a CPU implementing it in silicon, yet.

**Comment by fanghuaqi** [comment]
Yeah, there are some small customized silicon which not easy to buy directly are using rv32 linux, and also there is a [naxriscv](https://github.com/SpinalHDL/NaxRiscv?tab=readme-ov-file#running-verilator-simulation) fpga soft-core is riscv32, can boot linux, maybe rv32 patches for strace could be sent and merged for such use cases.

**Comment by rxrbln** [comment]
> Yeah, there are some small customized silicon which not easy to buy directly are using rv32 linux, and also there is a [naxriscv](https://github.com/SpinalHDL/NaxRiscv?tab=readme-ov-file#running-verilator-simulation) fpga soft-core is riscv32, can boot linux, maybe rv32 patches for strace could be sent and merged for such use cases.

I was answering to "riscv64 does support riscv32 compat", 32-bit only riscv32 silicon of course exists a lot. The patch was originally sent by Western Digital, idk why it was not merged. I can certainly ping it. But running https://t2linux.com for 25 I learned not to waste my time on upstreaming stuff upstream does not want ;-)

**Comment by fanghuaqi** [comment]
> > Yeah, there are some small customized silicon which not easy to buy directly are using rv32 linux, and also there is a [naxriscv](https://github.com/SpinalHDL/NaxRiscv?tab=readme-ov-file#running-verilator-simulation) fpga soft-core is riscv32, can boot linux, maybe rv32 patches for strace could be sent and merged for such use cases.
> 
> I was answering to "riscv64 does support riscv32 compat", 32-bit only riscv32 silicon of course exists a lot. The patch was originally sent by Western Digital, idk why it was not merged. I can certainly ping it. But running https://t2linux.com for 25 I learned not to waste my time on upstreaming stuff upstream does not want ;-)

Ooh, sorry, I misunderstand it, I hope maybe the strace maintainers @ldv-alt maybe can somehow take a look at this rv32 strace patch here https://t2sde.org/packages/strace.
